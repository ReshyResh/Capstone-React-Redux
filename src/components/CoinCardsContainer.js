import React from 'react'
import CoinCard from './CoinCard';
import { v4 as uuidv4 } from 'uuid';

const CoinCardsContainer = (props) => {
    const { coins } = props;
    return (
        <div className = "card-container">
            {coins.map ((coin) => (
                <CoinCard coin = { coin } key={uuidv4()} />
            ))}
        </div>
    )
}

export default CoinCardsContainer
