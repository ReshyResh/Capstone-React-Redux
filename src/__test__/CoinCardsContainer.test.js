import React from 'react';
import renderer from 'react-test-renderer';
import CoinCardsContainer from '../components/CoinCardsContainer'
import mock from './MockApiContainer';
import { BrowserRouter as Router } from "react-router-dom";

it('Coin cards container renders correctly', () => {
  const tree = renderer
    .create(<Router><CoinCardsContainer coins = { mock } /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});