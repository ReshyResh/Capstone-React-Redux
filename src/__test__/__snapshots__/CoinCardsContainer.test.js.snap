// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin cards container renders correctly 1`] = `
<div
  className="card-container"
>
  <a
    href="/coin/ETHBTC"
    onClick={[Function]}
  >
    <div
      className="coin-card green-bg"
    >
      <div
        className="top-card"
      >
        <h3
          className="coin-card-title"
        >
           
          ETHBTC
          <br />
          +3.677%
        </h3>
      </div>
      <span
        className="arrow-details"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </span>
      <img
        alt="ETHBTC"
        className="card-picture"
        src="Image not found"
      />
    </div>
  </a>
  <a
    href="/coin/LTCBTC"
    onClick={[Function]}
  >
    <div
      className="coin-card green-bg"
    >
      <div
        className="top-card"
      >
        <h3
          className="coin-card-title"
        >
           
          LTCBTC
          <br />
          +1.949%
        </h3>
      </div>
      <span
        className="arrow-details"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </span>
      <img
        alt="LTCBTC"
        className="card-picture"
        src="Image not found"
      />
    </div>
  </a>
</div>
`;
